include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -76,128 +73,149 @@ struct view_equality_fn
+ * in the extended CIGAR alphabet (`extended_cigar` = `true`) the function
  * will return an 'X' since the bases are aligned but are not
  * equal.
  * \sa seqan3::aligned_sequence
  */
 template <typename reference_char_type, typename query_char_type>
+[[nodiscard]] constexpr cigar_op map_aligned_values_to_cigar_op(reference_char_type const reference_char,
+                                                                query_char_type const query_char,
+                                                                bool const extended_cigar)
 //!\cond
     requires seqan3::detail::weakly_equality_comparable_with<reference_char_type, gap> &&
              seqan3::detail::weakly_equality_comparable_with<query_char_type, gap>
 //!\endcond
-char compare_aligned_values(reference_char_type const reference_char,
-                            query_char_type const query_char,
-                            bool const extended_cigar)
 {
-    return (reference_char == gap{})
-                ? (query_char == gap{})
-                    ? 'P'
-                    : 'I'
-                : (query_char == gap{})
-                    ? 'D'
-                    : (extended_cigar)
-                        ? (query_char == reference_char)
-                            ? '='
-                            : 'X'
-                        : 'M';
+    constexpr std::array<char, 6> operators{'M', 'D', 'I', 'P', 'X', '='};  // contains the possible cigar operators.
+    uint8_t key = (static_cast<uint8_t>(reference_char == gap{}) << 1) | static_cast<uint8_t>(query_char == gap{});
+    if (extended_cigar && (key == 0)) // in extended format refine the substitution operator to match/mismatch.
+        key |= ((1 << 2) | static_cast<uint8_t>(query_char == reference_char));  // maps to [4, 5].
+
+    return assign_char_to(operators[key], cigar_op{});
 }
 
-/*!\brief Transforms an alignment represented by two aligned sequences into the
- *        corresponding CIGAR string.
+/*!\brief Creates a CIGAR string (SAM format) given an alignment represented by two aligned sequences.
  * \ingroup alignment_file
  *
- * \tparam ref_seq_type    Must model std::ranges::forward_range. The value_type must
- *                         be equality comparable to seqan3::gap.
- * \tparam query_seq_type  Must model std::ranges::forward_range. The value_type must
- *                         be equality comparable to seqan3::gap.
- * \param  ref_seq         The reference sequence to compare against the query sequence.
- * \param  query_seq       The query sequence to build the CIGAR string for.
+ * \tparam alignment_type  Must model the seqan3::tuple_like and must have std::tuple_size 2.
+ *                         Each tuple element must be a range over values comparable to seqan3::gap.
+ * \param  alignment       The alignment, represented by a pair of aligned sequences,
+ *                         to be transformed into CIGAR_vector based on the
```

Comment:
```
CIGAR_vector
cigar_vector
cigar vector
```
From these three I would prefer the latter since it does not name a variable really. Also in the note block